packages/graphql/src/schema/resolvers/subscriptions/utils/compare-properties.ts

```diff
@@ -17,10 +17,161 @@
  * limitations under the License.
  */
 
+import { int } from "neo4j-driver";
+import type Node from "../../../../classes/Node";
+import type { PrimitiveField } from "../../../../types";
+import { whereRegEx } from "../../../../translate/where/utils";
+import type { WhereRegexGroups } from "../../../../translate/where/utils";
+
+const isObject = (x) => typeof x === "object" && !Array.isArray(x) && x !== null;
 /** Returns true if all properties in obj1 exists in obj2, false otherwise */
-export function compareProperties<T>(obj1: Record<string, T>, obj2: Record<string, T>): boolean {
+export function compareProperties<T>(obj1: Record<string, T> | T, obj2: Record<string, T> | T): boolean {
+    if (!isObject(obj1) && !Array.isArray(obj1)) {
+        // coming from: array of strings
+        return obj1 === obj2;
+    }
     for (const [k, value] of Object.entries(obj1)) {
-        if (obj2[k] !== value) {
+        if (Array.isArray(value)) {
+            if (obj2[k].length !== value.length) {
+                return false;
+            }
+            const compare = (other: T): boolean => {
+                // OR bc. when comparing arrays of arity > 1 for sure there will be non-matches.
+                // However, in the case of all non-matches, false will be returned.
+                return value.reduce((acc, v) => acc || compareProperties(v, other), false);
+            };
+            const valuesToCompareWith = Object.values(obj2[k] as ArrayLike<T>);
+            const noExpectingResults = Math.max(valuesToCompareWith.length, Object.values(value).length);
+            const noMatchingResults = valuesToCompareWith.map(compare).filter((x) => x).length;
+            if (noMatchingResults !== noExpectingResults) {
+                return false;
+            }
```

Comment:
I may be missing something here, but we have already checked that both arrays have the same length, so couldn't we just make the same logic as `filterByProperties` and loop until a `false` happens?

```ts
for(const value of valuesToCompareWith) {
    if(!compare(value)) {
        return false
    }
}
```

This logic seems a bit simpler. Another plus is that we are short-circuiting the comparison, returning at the first false instead of performing all of the comparisons. Thoughts :thinking: ?

Reply:
yep, I missed this :D it was "leftover" from a previous implementation. Good catch